analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ConditionEvaluatesToConstantBase.cs

```diff
@@ -112,11 +112,11 @@ private void ReportIssue(IOperation operation, BasicBlock block, bool conditionV
         var secondaryLocations = SecondaryLocations(block, conditionValue, syntax.Span.End);
         if (secondaryLocations.Any())
         {
-            ReportIssue(Rule2583, syntax, secondaryLocations, properties: null, issueMessage, S2583MessageSuffix);
+            ReportIssue(Rule2583, syntax, secondaryLocations.Select(x => x.ToSecondary()), issueMessage, S2583MessageSuffix);
         }
         else
         {
-            ReportIssue(Rule2589, syntax, additionalLocations: null, properties: null, issueMessage, string.Empty);
+            ReportIssue(Rule2589, syntax, [], issueMessage, string.Empty);
```

Comment:
This looks wrong. Can't we just remove the secondary location?
```suggestion
            ReportIssue(Rule2589, syntax, issueMessage, string.Empty);
```

Reply:
I tried, but it didn't work. We don't have overload with `Rule` parameter.